doc/source/whatsnew/v0.22.0.txt

```diff
@@ -60,7 +60,7 @@ Deprecations
 Removal of prior version deprecations/changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
--
+- The warnings for construction of a ``Categorical`` in the form ``Categorical(codes, categories)`` have been removed (:issue:`8074`)
```

Comment:
just tweak this a bit, saying that when codes is an integer dtype; say this is in favor of ``Categorical.from_codes``, otherwise lgtm. ping on green.